transmogrifier/models.py

```diff
@@ -216,3 +216,19 @@ class TimdexRecord:
 
     def asdict(self) -> dict[str, Any]:
         return asdict(self, filter=lambda _, value: value is not None)
+
+    @classmethod
+    def get_required_field_names(cls) -> list[str]:
+        return [
+            field
+            for field, attribute in attrs.fields_dict(cls).items()
+            if attribute.default is attrs.NOTHING
+        ]
```

Comment:
As discussed, I currently don't see a use for this but open to keeping it if a different approach to orchestration is identified during the review

Reply:
I'd still vote to keep it.  To me the value is symmetry (as you mentioned), and leaning into separation of concerns.  If we imagine for a moment we have no idea how these will be used, they still have values for unambiguously telling us which fields are requierd, and which are optional, for a TIMDEX record.   Having only optional fields odd in that context.

Of course, it turns out maybe we only use that method, but from the lens of the `TimdexRecord` class only, I think it makes good sense to have both.

Reply:
Let's keep it! As @ghukill , I think it's a useful method when considering the `TimdexRecord` class (i.e., at any point, we can call `TimdexRecord.get_required_field_names`).

Reply:
Sounds good!